src/SubtitlesParser/index.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Log } from '../LightningSdkPlugins'
+
+export default class SubtitlesParser {
+  // @ params url: subtitle file URL
+  // @ customParser: a customParser to use instead of default parser of the plugin
+  // @ parseOptions.removeSubtitleTextStyles: remove subtitle textstyles possible value true or false
+  // @return parsed subtitles as list of objects
+  // also stores parsed data
+  static fetchAndParseSubs(
+    url,
+    customParser = false,
+    parseOptions = { removeSubtitleTextStyles: true }
+  ) {
+    let _url
+
+    try {
+      _url = new URL(url)
+    } catch (e) {
+      Log.info('Invalid URL')
+      return Promise.reject(new Error('Invalid URL'))
+    }
+    if (!((_url.protocol === 'https:' || _url.protocol === 'http:') && _url.hostname)) {
+      Log.info('Invalid subtitle Url')
+      return Promise.reject(new Error('Invalid URL'))
+    }
+
+    return new Promise((resolve, reject) => {
+      fetch(url)
+        .then(data => data.text())
+        .then(subtitleData => {
+          this.clearAllSubtitles()
+          if (customParser && typeof customParser === 'function') {
+            this._captions = customParser(subtitleData)
+          } else {
+            this._captions = this.parseSubtitles(subtitleData, parseOptions)
+          }
+          if (!this._captions.length) {
```

Comment:
Why do you need this check? Technically, even when subtitles are successfully parsed but yield an empty array, they are not strictly invalid. A use case could be that a subtitle file has been created for all assets but need yet to be translated (a bit of a stretch but still). I wouldn't consider it a warning

Reply:
yes, but when the subtitles are parsed and he can't see subtitles on UI, this warning will help in understanding the actual issue is not with the parser, the file itself doesn't contain any subtitles.